### PR TITLE
Migrate algolia search to v3

### PR DESCRIPTION
**Pull request by paul-vd**
### What?
- Migrated the Algolia docsearch to v3 ([migration](https://docsearch.algolia.com/docs/migrating-from-v2/))
- Updated the crawler selectors to include category titles instead of collecting everything under a single `Documentation`
- Improved the font-weight in the sidebar for better text readability

https://user-images.githubusercontent.com/39598117/157044142-890bfc80-9c43-40a6-afc4-79bf9dcca652.mp4

~ fixes https://github.com/front-commerce/developers.front-commerce.com/pull/346#issuecomment-1058904478

### CODE REVIEW

**Comment by paul-vd** [comment]
I also updated the crawler induction to add category title

https://user-images.githubusercontent.com/39598117/157044142-890bfc80-9c43-40a6-afc4-79bf9dcca652.mp4

s

File: theme/layout/partials/sidebar.ejs [hunk 1]
```diff
@@ -69,31 +69,9 @@
...
-      <div class="input-area">
-        <div class="input-symbol small round">
-          <input type="text" placeholder="Search <%- config.subtitle %>" id="desktop-search-input" />
-          <span class="icon-search"></span>
-        </div>
-      </div>
-      <div class="wrapper-desktop-search-results"></div>
-    </div>
-
+      <div id="docsearch"></div>
```
Comment (real34) [inline theme/layout/partials/sidebar.ejs:74]: **nitpick:** removing the `.wrapper-search` removed some vertical spacing too.
I find the new spacing a bit too small.

screenshot-2022-03-07_18-58-58
screenshot-2022-03-07_18-57-22


**suggestion:** add more spacing below the search. In CSS or with
```suggestion
    <div class="wrapper-search">
        <div id="docsearch"></div>
    </div>
```

**Review by real34 — APPROVED** [review]
Thanks, it is way better!

I noticed a _regression_ from this config, for a person who wants to access to the migration guides. Could you check what is the config behind it?

Before:

screenshot-2022-03-07_19-06-06

After:

screenshot-2022-03-07_19-00-50

File: theme/layout/partials/sidebar.ejs (hunk 1 above)
Comment (paul-vd) [reply theme/layout/partials/sidebar.ejs:74]: > I noticed a regression from this config, for a person who wants to access to the migration guides. Could you check what is the config behind it?

I found that the new version is not successfully crawling the migration guide as it's complaining that there are too many records 🦘 
image
image

<br />
<br />

> nitpick: removing the .wrapper-search removed some vertical spacing too.
I find the new spacing a bit too small.

Nice catch! I'm going to call you hawkeye 🦅 from now on hehe I'll fix that.

Comment (paul-vd) [reply theme/layout/partials/sidebar.ejs:74]: I created an additional step when crawling the migration guide to only pull the tagged headers
image

Comment (paul-vd) [reply theme/layout/partials/sidebar.ejs:74]: Updated to spacing aswell

**Pull request merged**